samples/nrf9160/modem_shell/src/ltelc/ltelc.c

```diff
@@ -0,0 +1,505 @@
+					mem_slot_index, normal_mode_at_cmd,
+					response);
+			}
+		}
+	}
+
+	return 0;
+}
+
+void ltelc_rsrp_subscribe(bool subscribe)
+{
+	ltelc_subscribe_for_rsrp = subscribe;
+	if (shell_global != NULL) {
+		if (ltelc_subscribe_for_rsrp) {
+			/* print current value right away: */
+			shell_print(shell_global, "RSRP subscribed");
+			if (modem_rsrp != LTELC_RSRP_VALUE_NOT_KNOWN) {
+				shell_print(shell_global, "RSRP: %d", modem_rsrp);
+			}
+		} else {
+			shell_print(shell_global, "RSRP unsubscribed");
+		}
+	}
+}
+
+void ltelc_ncellmeas_start(bool start)
+{
+	int ret;
+
+	if (start) {
+		shell_print(
+			shell_global,
+			"Neighbor cell measurements and reporting starting");
+		ret = lte_lc_neighbor_cell_measurement();
+		if (shell_global != NULL) {
+			if (ret) {
+				shell_error(
+					shell_global,
+					"lte_lc_neighbor_cell_measurement() returned err %d",
+					ret);
+				shell_error(
+					shell_global,
+					"Cannot start neigbor measurements");
+			}
+		}
+	} else {
+		ret = lte_lc_neighbor_cell_measurement_cancel();
+		if (shell_global != NULL) {
+			if (ret) {
+				shell_error(
+					shell_global,
+					"lte_lc_neighbor_cell_measurement_cancel() returned err %d",
+					ret);
+			} else {
+				shell_print(
+					shell_global,
+					"Neighbor cell measurements and reporting stopped");
+			}
+		}
+	}
+}
+
+#define AT_MDM_SLEEP_NOTIF_START "AT%%XMODEMSLEEP=1,%d,%d"
+#define AT_MDM_SLEEP_NOTIF_STOP "AT%XMODEMSLEEP=0"
+void ltelc_modem_sleep_notifications_subscribe(uint32_t warn_time_ms,
+					       uint32_t threshold_ms)
+{
+	char buf_sub[48];
+	int err;
+
+	snprintk(buf_sub, sizeof(buf_sub), AT_MDM_SLEEP_NOTIF_START,
+		 warn_time_ms, threshold_ms);
+
+	err = at_cmd_write(buf_sub, NULL, 0, NULL);
+	if (err) {
+		shell_error(
+			shell_global,
+			"Cannot subscribe to modem sleep notifications, err %d",
+			err);
+	} else {
+		shell_print(shell_global,
+			    "Subscribed to modem sleep notifications");
+	}
+}
+
+void ltelc_modem_sleep_notifications_unsubscribe(void)
+{
+	int err;
+
+	err = at_cmd_write(AT_MDM_SLEEP_NOTIF_STOP, NULL, 0, NULL);
+	if (err) {
+		shell_error(shell_global,
+			    "Cannot stop modem sleep notifications, err %d",
+			    err);
+	} else {
+		shell_print(shell_global,
+			    "Unsubscribed from modem sleep notifications");
+	}
+}
+
+#define AT_TAU_NOTIF_START "AT%%XT3412=1,%d,%d"
+#define AT_TAU_NOTIF_STOP "AT%%T3412=0"
+void ltelc_modem_tau_notifications_subscribe(uint32_t warn_time_ms,
+					     uint32_t threshold_ms)
+{
+	char buf_sub[48];
+	int err;
+
+	snprintk(buf_sub, sizeof(buf_sub), AT_TAU_NOTIF_START, warn_time_ms,
+		 threshold_ms);
+
+	err = at_cmd_write(buf_sub, NULL, 0, NULL);
+	if (err) {
+		shell_error(shell_global,
+			    "Cannot subscribe to TAU notifications, err %d",
+			    err);
+	} else {
+		shell_print(shell_global, "Subscribed to TAU notifications");
+	}
+}
+
+void ltelc_modem_tau_notifications_unsubscribe(void)
+{
```

Comment:
To be fixed, seems to be copy-paste from ltelc_modem_sleep_notifications_unsubscribe()